x-pack/plugins/enterprise_search/server/routes/enterprise_search/indices.ts

```diff
@@ -141,25 +133,21 @@ export function registerIndexRoutes({ router }: RouteDependencies) {
         }),
       },
     },
-    async (context, request, response) => {
+    elasticsearchErrorHandler(log, async (context, request, response) => {
       const { indexName } = request.params;
       const { client } = (await context.core).elasticsearch;
-      try {
-        const indexExists = await client.asCurrentUser.indices.exists({ index: indexName });
-        return response.ok({
-          body: {
-            exists: indexExists,
-          },
-          headers: { 'content-type': 'application/json' },
-        });
-      } catch (error) {
-        return response.customError({
-          body: 'Error fetching data from Enterprise Search',
-          statusCode: 502,
-        });
-      }
-    }
+
+      const indexExists = await client.asCurrentUser.indices.exists({ index: indexName });
```

Comment:
I think this route we want to catch and log the error and just return false with a 200, because I wouldn't want to bother the user with unauthorized errors every time they type in (part of) an index name they don't have access to.

Reply:
👍